src/ui_plugin/systems/search.rs

```diff
@@ -91,6 +95,12 @@ pub fn search_box_text_changed(
                 let result = fuzzy_search(index, str.as_str());
                 match result {
                     Ok(docs) => {
+                        let mut doc_ids = docs
+                            .clone()
+                            .into_iter()
+                            .map(|l| ReflectableUuid(l.node_id))
+                            .collect::<Vec<_>>();
```

Comment:
Let's use `std::collections::HashSet` to have O(1) amortised contains operation.

Reply:
Something like this:
```rust
let node_ids: HashSet<ReflectableUuid> = docs
    .into_iter()
    .filter(|l| l.doc_id == current_doc_id)
    .map(|l| ReflectableUuid(l.node_id))
    .collect()
```